py/server/tests/test_data_index.py

```diff
@@ -0,0 +1,44 @@
+#
+# Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+#
+
+import unittest
+
+from deephaven import empty_table
+from deephaven.experimental.data_index import create_data_index, has_data_index, get_data_index
+from tests.testbase import BaseTestCase
+
+class DataIndexTestCase(BaseTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.table = empty_table(10).update(["Timestamp = now()", "X=i%3", "Y=`Deephaven` + String.valueOf(ii)", "Z=ii*2"])
+        self.data_index = create_data_index(self.table, ["X", "Y"])
+        self.table_nodi = empty_table(10).update(["Timestamp = now()", "X=i%3", "Y=`Deephaven` + String.valueOf(ii)",
+                                           "Z=ii*2"])
+
+    def test_has_data_index(self):
+        self.assertTrue(has_data_index(self.table, ["X", "Y"]))
+        self.assertFalse(has_data_index(self.table, ["X"]))
+        self.assertFalse(has_data_index(self.table, ["X", "Z"]))
+        self.assertFalse(has_data_index(self.table_nodi, ["X", "Y"]))
+        self.assertFalse(has_data_index(self.table_nodi, ["X"]))
+
+    def test_get_data_index(self):
+        self.assertIsNotNone(get_data_index(self.table, ["X", "Y"]))
+        self.assertIsNone(get_data_index(self.table, ["X"]))
+        self.assertIsNone(get_data_index(self.table, ["X", "Z"]))
+        self.assertIsNone(get_data_index(self.table_nodi, ["X", "Y"]))
+
+    def test_keys(self):
+        self.assertEqual(["X", "Y"], self.data_index.keys)
+
+    def test_backing_table(self):
+        self.assertEqual(3, len(self.data_index.table.columns))
+        self.assertEqual(10, self.data_index.table.size)
+        di = create_data_index(self.data_index.table, self.data_index.keys[0:1])
```

Comment:
This method should have a stand alone test that carefully checks both "get" and "create" and "bad columns"

Reply:
This test looks hacky -- unless I'm misreading it.  This looks like it is indexing the index table for the sake of a unit test.

Reply:
It is written to test that the backing table is truly just a table and it too can be indexed.